src/core/stores/VDocument.ts

```diff
@@ -38,6 +38,34 @@ export class VDocument {
         this.range.start.parent.after(duplicatedParent);
         nextSiblings.forEach(sibling => duplicatedParent.append(sibling));
     }
+    /**
+     * Change the selection's paragraph formatting. If there is no selection,
+     * first select the parent of the range nodes.
+     *
+     * Examples:
+     *
+     * - `<paragraph>te◆xt</paragraph>` =>
+     *   `changeType(VNodeType.HEADING1)` =>
+     *   `<heading1>▶text◀</heading1>`
+     *
+     * - `<heading1>te▶xt</heading1><heading2>te◀xt</heading2>`=>
+     *   `changeType(VNodeType.PARAGRAPH)` =>
+     *   `<paragraph>te▶xt</paragraph><paragraph>te◀xt</paragraph>`
+     *
+     * @param type
+     */
+    formatParagraph(type: VNodeType): void {
+        if (this.range.isCollapsed()) {
+            this.range.select(this.range.start.parent, this.range.end.parent);
```

Comment:
On VRange then? Like `selectParent` or `selectUp` ? That could be useful. Maybe wait to see if we need it a second time?

Reply:
Maybe that might be the default behavior of `select` ? You'd do `this.range.select()` and that would do that. We already have a way to express "select nothing" with `this.range.collapse()`, so it might make sense that `select` *always* select something, and this would be the default behavior if you don't specify *what* to select.